cartography/intel/aws/route53.py

```diff
@@ -9,6 +9,7 @@ def link_aws_resources(neo4j_session, update_tag):
     # find records that point to other records
     link_records = """
     MATCH (n:AWSDNSRecord) WITH n MATCH (v:AWSDNSRecord{value: n.name})
+    WHERE NOT n = v
```

Comment:
I would be more specific here.
e.g. `WHERE n.value <> v.value` or whatever the correct way of identifying them is.

Reply:
I want to make sure the node is not equal to the other node, not just making sure a property is not the same. This was causing self-relationships